### PR TITLE
Reckoner v1.0.1

### PR DESCRIPTION
**Pull request by endzyme**
Fixes
* #60 - Better post hook experience with relative pathing to course.yml
* #59 - Better visibility into post hooks
* #18 - Resolved missing heading definitions with better error handling

### CODE REVIEW

**Comment by endzyme** [comment]
NOTE: I will be squash merging these due to the long tail of commits here.

**Comment by sudermanjr** [comment]
I'm getting some errors trying to run anything with hooks.

Snippet of course.yaml
```yaml
  metrics-server:
    namespace: metrics-server
    version: "2.4.0"
    hooks:
      pre_install:
        - kubectl create ns metrics-server || true
        - kubectl annotate ns metrics-server --overwrite linkerd.io/inject=enabled
    set-values:
      args:
        - --kubelet-insecure-tls
        - --kubelet-preferred-address-types=InternalIP
        - --metric-resolution=30s
        - --v=3
```

Output:
```
2019-02-15 16:19:58 root[12435]     INFO Installing metrics-server
2019-02-15 16:19:58 root[12435]     INFO Running pre_install hook...
2019-02-15 16:19:58 root[12435]    DEBUG kubectl create ns metrics-server || true
2019-02-15 16:19:58 root[12435]    ERROR [Errno 2] No such file or directory: ''
2019-02-15 16:19:58 root[12435]    ERROR None
2019-02-15 16:19:58 root[12435]    ERROR Helm upgrade failed on metrics-server
2019-02-15 16:19:58 root[12435]    DEBUG Traceback (most recent call last):
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/course.py", line 134, in plot
    chart.install(namespace=self.namespace, context=self.context)
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/chart.py", line 220, in install
    self.pre_install_hook()
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/chart.py", line 126, in pre_install_hook
    self.run_hook('pre_install')
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/chart.py", line 157, in run_hook
    "'{}'".format(command)
ReckonerCommandException: Uncaught exception while running hook 'kubectl create ns metrics-server || true'

2019-02-15 16:19:58 root[12435]    ERROR ERROR: Some charts failed to install.
2019-02-15 16:19:58 root[12435]    ERROR  - metrics-server
```

File: reckoner/tests/test_chart.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            }
+            },
+            None,
+        )
+        chart.config.dryrun = False
+        chart.config.local_development = False
+
+        return chart
+
+    def test_execution_directory(self, mock_cmd_call, *args):
```
Comment (endzyme) [inline reckoner/tests/test_chart.py]: This needs a test to verify we can run in the same dir as course.yaml

**Comment by ejether** [comment]
I just noticed that `./tests` is duplicated into `./reckoner/tests` which means it would be delivered in the package. We probably don't want that. 

Also, and probably my fault, `setup.py` line 42, exclude is `test` and should be `tests`.

**Comment by endzyme** [comment]
> I just noticed that `./tests` is duplicated into `./reckoner/tests` which means it would be delivered in the package. We probably don't want that.
> 
> Also, and probably my fault, `setup.py` line 42, exclude is `test` and should be `tests`.

This packaging issue is resolved 

I think we're ready for a merge now

**Pull request merged**